### ISSUE TITLE
[Performance]: Clarification on Base Model Inference Count with Multiple LoRA Models in vLLM Deployment

### ISSUE THREAD

**Issue by zhangyuqi-1** [vllm-project/vllm#8228]
### Misc discussion on performance

Question:

When deploying LoRA with vLLM, suppose I have 1000 different LoRA models, and each LoRA receives a separate request with a different input. In this scenario, how many times does the base model actually perform inference? Is it only once, or does it perform 1000 inferences?

I understand that the LoRA part will run 1000 times, but its computational cost is relatively small. I'm mainly concerned about how many times the base model runs inference in this case. If the base model only runs once, that would be incredibly efficient, meaning that as the number of LoRA models increases, the overall efficiency would improve significantly. Is this possible?

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by jeejeelee** [comment]
> When deploying LoRA with vLLM, suppose I have 1000 different LoRA models, and each LoRA receives a separate request with a different input. In this scenario, how many times does the base model actually perform inference? Is it only once, or does it perform 1000 inferences?

Only once. If you want to delve deeper, see: https://github.com/vllm-project/vllm/pull/1804

**Comment by zhangyuqi-1** [comment]
It’s fascinating! Does it batch requests corresponding to different LoRA models into one batch size, allowing it to only perform inference once? I’m curious how this is achieved.

**Comment by jeejeelee** [comment]
> It’s fascinating! Does it batch requests corresponding to different LoRA models into one batch size, allowing it to only perform inference once? I’m curious how this is achieved.

See:  [punica paper](https://arxiv.org/abs/2310.18547)  or  it's [blog](https://le.qun.ch/en/blog/2023/09/11/multi-lora-potentials/)

**Comment by zhangyuqi-1** [comment]
> > It’s fascinating! Does it batch requests corresponding to different LoRA models into one batch size, allowing it to only perform inference once? I’m curious how this is achieved.
> 
> See: [punica paper](https://arxiv.org/abs/2310.18547) or it's [blog](https://le.qun.ch/en/blog/2023/09/11/multi-lora-potentials/)

thanks!
